FIRO_TSEnsembles/src/main/java/hec/stats/InlineStats.java

```diff
@@ -38,8 +43,12 @@ public void AddObservation(float value){
                 _min = value;
             }
             _sampleSize +=1;
-            _sampleVariance = (float) ((_sampleSize -2)/(_sampleSize-1))*_sampleVariance + (float)java.lang.Math.pow(value-_mean,2)/(float)(_sampleSize);
+
+
+            _sampleVariance = ((float)(_sampleSize - 2) / (_sampleSize - 1)) * _sampleVariance + (float) Math.pow(value-_mean,2)/(float)(_sampleSize);
             _mean += ((value - _mean)/(float)(_sampleSize));
+
         }
+        _standardDeviation = (float) Math.sqrt(_sampleVariance);
```

Comment:
my reccomendation is to not do it this way. Now every time you add an observation you call sqrt.

You only need to call sqrt when someone requests variance.

Reply:
updated based on recommendation